app/ui/legacy/src/main/java/com/fsck/k9/account/AccountRemoverWorker.kt

```diff
@@ -0,0 +1,48 @@
+package com.fsck.k9.account
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.Data
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+/**
+ * An [AccountRemoverWorker] to remove an account in the background.
+ */
+class AccountRemoverWorker(
+    context: Context,
+    workerParams: WorkerParameters,
+) : CoroutineWorker(context, workerParams), KoinComponent {
+
+    private val accountRemover: AccountRemover by inject()
+
+    override suspend fun doWork(): Result = withContext(Dispatchers.IO) {
+        val accountUuid = inputData.getString(ARG_ACCOUNT_UUID)
+            ?: throw IllegalArgumentException("No account UUID provided")
+
+        accountRemover.removeAccount(accountUuid)
+
+        Result.success()
+    }
+
+    companion object {
+        private const val ARG_ACCOUNT_UUID = "accountUuid"
+
+        fun enqueueRemoveAccountJob(context: Context, accountUuid: String) {
+            val data = Data.Builder()
+                .putString(ARG_ACCOUNT_UUID, accountUuid)
+                .build()
+
+            val request = OneTimeWorkRequest.Builder(AccountRemoverWorker::class.java)
```

Comment:
This should be expedited work. See https://developer.android.com/guide/background/persistent/getting-started/define-work#expedited